traitlets/traitlets.py

```diff
@@ -356,13 +357,14 @@ def __init__(self, default_value=Undefined, allow_none=None, **metadata):
             )
 
         if len(metadata) > 0:
+            warn("metadata should be set by ...", DeprecationWarning, stacklevel=2)
```

Comment:
Let's reach consensus on the discussion in #52 before going too much
further on the implementation.

On Fri, Jul 17, 2015 at 12:48 PM, Jason Grout <notifications@github.com>
wrote:

> In traitlets/traitlets.py
> <https://github.com/ipython/traitlets/pull/53#discussion_r34924703>:
>
> > @@ -356,13 +357,14 @@ def __init__(self, default_value=Undefined, allow_none=None, **metadata):
> >              )
> >
> >          if len(metadata) > 0:
> > +            warn("metadata should be set by ...", DeprecationWarning, stacklevel=2)
>
> Fix this message with whatever is decided
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipython/traitlets/pull/53/files#r34924703>.
>



-- 
Brian E. Granger
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com